apis/addons/v1alpha1/addons_types.go

```diff
@@ -86,6 +87,35 @@ type MonitoringSpec struct {
 	// and it needs to be runing inside the namespace specified by `.monitoring.federation.namespace`
 	// with the service name 'prometheus'.
 	Federation *MonitoringFederationSpec `json:"federation,omitempty"`
+
+	// Settings For Monitoring Stack
+	// +optional
+	MonitoringStack *MonitoringStackSpec `json:"monitoringStack,omitempty"`
+}
+
+type MonitoringStackSpec struct {
+
+	// Enabled when set to true, deploys a MonitoringStack object
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
I would drop this in favor of just setting `.spec.monitoringStack = null` or we have two ways to disable the monitoring stack.